src/fasta-recs.c

```diff
@@ -9,8 +40,63 @@ int main(int argc, char const *argv[])
     }
 
     const char *fasta_fname = argv[1];
-    printf("I should be parsing %s now, but someone forgot to implement that.\n",
-           fasta_fname);
+
+    char *content = readFile(fasta_fname);
+
+    //Parsing into DNA chains
+    int pos = 0;
+    const int length = strlen(content);
+
+    // Find every chain
+    while(pos<length){
+        //First we have to find '>'
+        char c = content[pos];
+        while(pos < length && c!='>'){
+            pos++;
+            c = content[pos];
+        }
+
+        // Skip spaces
+        while(pos < length && c!=' '){
+            pos++;
+            c = content[pos];
+        }
+
+        // Find the name
+        int nameBegin = pos;
+        while(pos<length && c!=' ' && c!='\n'){
+            pos++;
+            c = content[pos];
+        }
+
+        char *name = calloc(pos-nameBegin, sizeof(char));
+        memcpy(name, &content[nameBegin], pos);
+
+        // Find the chain
+        int nChain = 0;
+        int chainBegin = pos;
+        while(pos<length && c!='>'){
+            if(c!=' ' && c!='\n'){
+                nChain++;
+            }
+
+            pos++;
+            c = content[pos];
+        }
+
+        printf("%s ",name);
+
+        for(int i=0; i<pos-chainBegin && nChain>0; i++){
+            char c = content[chainBegin+i];
+            if(c!=' ' && c!='\n'){
+                printf("%c", c);
+                nChain--;
+            }
+        }
+
```

Comment:
I suppose this is one way of doing it... what I had in mind was to extract the relevant sequences, since that is what you need later, but you will of course pass the tests if you can just write them to stdout. It won't be sufficient later on, however.